### PR TITLE
LPS-168542 & LPS-168525 Improve accessibility in membership and organization tabs

### PR DESCRIPTION
**Pull request by veroglez**
This pull request solves the following issues:

https://issues.liferay.com/browse/LPS-168525
https://issues.liferay.com/browse/LPS-168542

### CODE REVIEW

**Comment by veroglez** [comment]
ci:test:sf

**Comment by veroglez** [comment]
ci:test:sf

**Comment by veroglez** [comment]
ci:test:relevant

**Comment by veroglez** [comment]
ci:test:stable

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp [hunk 1]
```diff
@@ -201,18 +210,29 @@ currentURLObj.setParameter("historyKey", liferayPortletResponse.getNamespace() +
 				Util.openSelectionModal({
 					onSelect: (selectedItem) => {
 						if (selectedItem) {
-							const entityId = selectedItem.entityid;
-
+							const {
+								entityid: entityId,
+								entityname: entityName,
+								type,
+							} = selectedItem;
```
Comment (drewbrokke) [inline modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp:217]: Is destructuring supported on JSP pages? I only see one other usage like this in portal.

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp [hunk 2]
```diff
@@ -201,18 +210,29 @@ currentURLObj.setParameter("historyKey", liferayPortletResponse.getNamespace() +
...
 
-							rowColumns.push(selectedItem.entityname);
-							rowColumns.push(selectedItem.type);
+							let removeButton =
+								'<%= UnicodeFormatter.toString(removeButtonOrganizations) %>';
+
+							removeButton = removeButton
+								.replace('aria-label=""', `aria-label="\${label}"`)
+								.replace('data-rowId=""', `data-rowId="\${entityId}"`)
+								.replace('title=""', `title="\${label}"`);
```
Comment (drewbrokke) [inline modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp:230]: For all these attribute replacements, could we instead use explicit token strings like TOKEN_ARIA_LABEL or something like it, then just do a direct replacement of the tokens? IMO that will make it a bit more obvious what's going on in these spots and easier to trace.

Something like:
```js
removeButton = removeButton
	.replace('TOKEN_ARIA_LABEL', label)
	.replace('TOKEN_DATA_ROW_ID', entityId)
	.replace('TOKEN_TITLE', label);
```

**Review by drewbrokke — CHANGES_REQUESTED** [review]
Overall looks good, just the one question and one suggestion. Thanks!

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp (hunk 1 above)
Comment (veroglez) [reply modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp:217]: Yes, there is no problem. It' a feature from ES6 and is supported in the portal. Other modules use it without problem.

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp (hunk 2 above)
Comment (veroglez) [reply modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp]: Thanks for the feedback! I like the idea, I update it.

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp (hunk 1 above)
Comment (drewbrokke) [reply modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp:217]: Makes sense - I've definitely seen it (and used it myself) in dedicated `js` files, but there's a few things that are legal in `js` files but are not legal in `jsp` files. Just wanted to double-check. Thanks @veroglez.

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp [hunk 3]
```diff
@@ -201,18 +210,29 @@ currentURLObj.setParameter("historyKey", liferayPortletResponse.getNamespace() +
 				Util.openSelectionModal({
 					onSelect: (selectedItem) => {
 						if (selectedItem) {
-							const entityId = selectedItem.entityid;
-
+							const {
+								entityid: entityId,
+								entityname: entityName,
```
Comment (drewbrokke) [inline modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp:215]: The `entityName` was being escaped in other files, but not here. Is that intentional?

**Review by drewbrokke — CHANGES_REQUESTED** [review]
Just the one question, otherwise this looks good. Thanks!

File: modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp (hunk 3 above)
Comment (veroglez) [reply modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp:215]: Yes, it's intentional. The previous code didn't escape the `entityName`. Do you think it's better to escape it?

Comment (drewbrokke) [reply modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp:215]: If it actually is a vector for XSS, then we should. If it's not a danger then we don't need to. Would you be willing to test that out?

Comment (veroglez) [reply modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp]: Yes, tested! the safest thing is to escape it. Thanks!

Comment (drewbrokke) [reply modules/apps/users-admin/users-admin-web/src/main/resources/META-INF/resources/user/organizations.jsp]: Great, thanks for testing that!

File: modules/apps/depot/depot-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.users.admin.web/user/memberships/depot_groups.jsp [hunk 4]
```diff
@@ -136,21 +146,28 @@ currentURLObj.setParameter("historyKey", liferayPortletResponse.getNamespace() +
 			Liferay.Util.openSelectionModal({
 				onSelect: function (selectedItem) {
 					if (selectedItem) {
-						var itemValue = JSON.parse(selectedItem.value);
+						const itemValue = JSON.parse(selectedItem.value);
+						const label = Liferay.Util.sub(
+							'<liferay-ui:message key="remove-x" />',
+							itemValue.title
```
Comment (drewbrokke) [inline modules/apps/depot/depot-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.users.admin.web/user/memberships/depot_groups.jsp:152]: @veroglez Is this also a possible XSS vector?

**Review by drewbrokke — CHANGES_REQUESTED** [review]

File: modules/apps/depot/depot-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.users.admin.web/user/memberships/depot_groups.jsp (hunk 4 above)
Comment (veroglez) [reply modules/apps/depot/depot-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.users.admin.web/user/memberships/depot_groups.jsp:152]: Here it is not necessary, it seems that the value is already escaped.

**Review by drewbrokke — APPROVED** [review]

**Comment by drewbrokke** [comment]
ci:forward

File: modules/apps/depot/depot-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.users.admin.web/user/memberships/depot_groups.jsp (hunk 4 above)
Comment (drewbrokke) [reply modules/apps/depot/depot-web/src/main/resources/META-INF/resources/dynamic_include/com.liferay.users.admin.web/user/memberships/depot_groups.jsp:152]: Sounds good, thanks for checking!

**Comment by drewbrokke** [comment]
@patriciajeanperez are there any related failures in this pull?

**Comment by patriciajeanperez** [comment]
I can send this with the test fixes @drewbrokke @eroglez

**Comment by drewbrokke** [comment]
Thanks @patriciajeanperez!

**Comment by patriciajeanperez** [comment]
ci:test:users

**Comment by patriciajeanperez** [comment]
ci:stop

**Comment by patriciajeanperez** [comment]
Pushed to BChan
https://github.com/brianchandotcom/liferay-portal/pull/131181

**Pull request closed without merging**